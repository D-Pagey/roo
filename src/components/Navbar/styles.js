import styled from 'styled-components';

export const Wrapper = styled.div`
    align-items: center;
    display: flex;
    justify-content: space-between;
    margin: 0 16px;
    height: 48px;
`;

export const UserWrapper = styled.div`
    align-items: center;
    display: flex;
`;

export const UserName = styled.span`
    margin: 0 0 0 4px;
`;
