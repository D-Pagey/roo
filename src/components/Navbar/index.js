import React from 'react';
import { useMediaQuery } from 'react-responsive';
import { mediaQuery } from '../../tokens';
import Logo from './assets/logo-horizontal.svg';
import SmallLogo from './assets/small-logo.svg';
import UserIcon from './assets/user-icon.svg';
import * as S from './styles';

const Navbar = () => {
    const isLaptopOrLarger = useMediaQuery({ query: mediaQuery.laptop });

    return (
        <S.Wrapper>
            {isLaptopOrLarger ? (
                <img src={Logo} alt="deliveroo logo" data-testid="navbarLogo" />
            ) : (
                <img src={SmallLogo} alt="deliveroo logo" data-testid="navbarSmallLogo" />
            )}

            <S.UserWrapper>
                <img src={UserIcon} alt="user icon" />
                {isLaptopOrLarger && <S.UserName>Jane Smith</S.UserName>}
            </S.UserWrapper>
        </S.Wrapper>
    );
};

export default Navbar;
