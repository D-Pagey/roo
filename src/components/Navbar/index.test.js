import React from 'react';
import { render } from '@testing-library/react';
import { useMediaQuery } from 'react-responsive';
import Navbar from '.';

jest.mock('react-responsive', () => ({
    useMediaQuery: jest.fn(() => false)
}));

describe('Navbar component', () => {
    it('should render', () => {
        const { container } = render(<Navbar />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render with small logo on mobile devices', () => {
        useMediaQuery.mockImplementationOnce(() => false);

        const { getByTestId } = render(<Navbar />);
        getByTestId('navbarSmallLogo');
    });

    it('should render with large logo of non-mobile devices', () => {
        useMediaQuery.mockImplementationOnce(() => true);

        const { getByTestId } = render(<Navbar />);
        getByTestId('navbarLogo');
    });
});
