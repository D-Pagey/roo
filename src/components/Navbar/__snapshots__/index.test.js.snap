// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 48px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 1126px;
  padding: 0 1rem;
  width: 100%;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0"
>
  <img
    alt="deliveroo logo"
    data-testid="navbarSmallLogo"
    src="small-logo.svg"
  />
  <div
    class="c1"
  >
    <img
      alt="user icon"
      src="user-icon.svg"
    />
  </div>
</div>
`;
